lms/djangoapps/grades/tasks.py

```diff
@@ -0,0 +1,54 @@
+"""
+This module contains tasks for asynchronous execution of grade updates.
+"""
+
+from celery import task
+from django.contrib.auth.models import User
+
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from lms.djangoapps.courseware.courses import get_course_by_id
+from opaque_keys.edx.keys import UsageKey
+from opaque_keys.edx.locator import CourseLocator
+from openedx.core.djangoapps.content.block_structure.api import get_course_in_cache
+
+from .config.models import PersistentGradesEnabledFlag
+from .transformer import GradesTransformer
+from .new.subsection_grade import SubsectionGradeFactory
+
+
+@task()
+def recalculate_subsection_grade(user_id, course_id, usage_id):
```

Comment:
I had intended to have separate async tasks for updating subsection and course grade tasks, with a successful subsection update then firing a signal to update the course grade as well. This extra granularity will allow for course grades to be updated independently (say, in response to a weighting change event).

Is this desirable, or are we intending the grade update task to be a single monolithic operation?